### DISCUSSION TITLE
ESBMC-AI

### DISCUSSION THREAD

**Discussion by Wu-kaixin** [esbmc/esbmc-ai#111] [Q&A] [answered]
Hi yiannis,

Hi yiannis, I am a master student of Prof. Lucas and I chose to study his ESBMC-AI framework.  so I was wondering if I could ask you some questions on this topic.

After I successfully downloaded and deployed the latest version of ESBMC and the ESBMC-AI framework, I wasn't able to operate as you did in the youtube video. My output: none. i don't know why this is happening.

Warm regards,
kaixin

**Comment by Yiannis128** [comment]
Hi, can you send a picture? Also, did you read the [Initial Setup](https://github.com/Yiannis128/esbmc-ai/wiki/Initial-Setup) instructions in the wiki?

**Comment by Wu-kaixin** [comment]
Yes I followed the tutorial. It's all configured too. Attached is a picture of mine.[
![屏幕截图 2024-03-10 133322](https://github.com/Yiannis128/esbmc-ai/assets/108617062/780dd018-c03b-43b9-8656-9d5014ab4b87)
](url)

**Comment by Wu-kaixin** [comment]
Maybe I don't know enough about frames, I don't know how you managed to get to that step inside your youtube video.
![屏幕截图 2024-03-10 133322](https://github.com/Yiannis128/esbmc-ai/assets/108617062/75757765-638d-4fb5-b997-fafb3edfa36c)

**Comment by Yiannis128** [comment]
Do you mind sending me the file you're trying to debug? I can try, and it would help resolve the problem faster.

**Reply by Wu-kaixin** [reply]
To be precise, my test script isn't finished yet. Just ran into these problems. Can you send me a script that will test the code? I'm not sure that my code is able to call the functions of the esbmc-ai framework.

**Reply by Wu-kaixin** [reply]
![屏幕截图 2024-03-11 184238](https://github.com/Yiannis128/esbmc-ai/assets/108617062/daf24cef-2190-40b9-8c1c-813bb9595934)
Now it seems I just don't know how this step works. It's that esbmc_ai.py script.

**Reply by Wu-kaixin** [reply]
The tool I use is pycharm

**Reply by Wu-kaixin** [reply]
I actually kind of don't quite understand what the esbmc_ai.py script does. Isn't it possible to call the framework directly via the [esbmc-ai samples/optimise-code/mult.c] directive to inspect the program?

**Reply by Yiannis128** [reply]
You can, but you would need to install it globally to call it as `./esbmc-ai`, instead on Linux/macOS call: `./esbmc-ai` and on Windows you can do `python3 -m esbmc_ai` to launch it.

**Comment by Wu-kaixin** [comment, marked as answer]
![屏幕截图 2024-03-11 185901](https://github.com/Yiannis128/esbmc-ai/assets/108617062/a4db4726-4208-4f3b-92ac-11488d173fb5)
I feel as if I've solved the problem. Right now there is just no quota for openai company APIs

**Reply by Yiannis128** [reply]
I think that you may need to either buy some credits to use OpenAI LLMs or simply switch to an open source alternative like Falcon :)

**Comment by Wu-kaixin** [comment]
I have purchased llm from openai. having some problems with the optimise code step.
![1](https://github.com/Yiannis128/esbmc-ai/assets/108617062/f1299ec4-0cc0-4c97-a20f-af3220fb18a0)
![2](https://github.com/Yiannis128/esbmc-ai/assets/108617062/30a08879-91bf-4484-930b-d72e344b7c7f)

**Reply by Wu-kaixin** [reply]
esbmc-ai samples/optimize-code/mult.c -I "C:\Users\kkk\PycharmProjects\llvm-mingw-20231128-ucrt-x86_64\include"
This directive is used because it keeps prompting for a <assert.h> header file that cannot be found.

**Reply by Wu-kaixin** [reply]
So what should be done next to optimise this framework and what needs to be done.

**Reply by Yiannis128** [reply]
Optimize code has been removed as it was very unstable. I advise against using it until a better implementation is complete. However, it seems that you don't have your C standard libraries installed, if `assert.h` is not found.

**Reply by Wu-kaixin** [reply]
I've already installed it. But I don't know why I can only find the location by assigning a value. I've also added it to the environment variable, but highlighting the incloud path, he still fails to find the location of the standard library.
![屏幕截图 2024-03-13 212020](https://github.com/Yiannis128/esbmc-ai/assets/108617062/78f72110-6cab-4452-aab4-3ed823593ab4)

**Reply by Yiannis128** [reply]
Does this work with ESBMC only?

**Reply by Wu-kaixin** [reply]
I don't think so. I think there may be a problem somewhere in the configuration of the environment that prevents the location of the standard library from being automatically located. But I have deployed one location of the standard library in the environment variables, so it doesn't make sense that it would have any effect on the other locations of the standard library (this part I think may be caused by the fact that I've downloaded and installed the standard library several times, so I've forgotten what folder some of the locations are in, and I feel like maybe it's a duplication problem).

**Comment by Wu-kaixin** [comment]
Hi, yiannis.How to use this framework further next to consider the detection and repair of errors in the C++ programming language. I made a script to summarise the output using a group of 10 C programs. The framework is further updated for multiple sets of output reports. Do you think this is correct?
